backend/services/implementations/testSessionService.ts

```diff
@@ -334,10 +334,41 @@ class TestSessionService implements ITestSessionService {
   async updateTestSession(
     id: string,
     testSession: TestSessionRequestDTO,
+    nowDate?: Date,
   ): Promise<TestSessionResponseDTO> {
     let updatedTestSession: TestSession | null;
 
     try {
+      const oldTestSession: TestSession | null = await MgTestSession.findById(
+        id,
+      );
+      if (!oldTestSession) {
+        throw new Error(`Test Session id ${id} not found`);
+      }
+
+      const date = nowDate ?? new Date();
```

Comment:
Could you move this to the top of the function, before the query? If the backend is slow and a teacher makes a modification 1s before the deadline they might expect it to go through.